jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -150,32 +149,54 @@ static Resource combine(Resource... resources)
     Resource newResource(URI uri);
 
     /**
-     * <p>Construct a system resource from a string.</p>
+     * <p>Construct a Resource from a string reference into classloaders.</p>
+     *
+     * @param resource Resource as string representation
+     * @return The new Resource, or null if string points to a location that does not exist
+     * @throws IllegalArgumentException if string is blank
+     * @see #newClassLoaderResource(String)
+     * @deprecated use {@link #newClassLoaderResource(String)} or {@link #newClassLoaderResource(String, boolean)} instead, will be removed in Jetty 12.1.0
+     */
+    @Deprecated(since = "12.0.1", forRemoval = true)
+    default Resource newSystemResource(String resource)
+    {
+        return newClassLoaderResource(resource);
+    }
+
+    /**
+     * <p>Construct a Resource from a search of ClassLoaders.</p>
      *
      * <p>
-     * The resource is first attempted to be accessed via the {@link Thread#getContextClassLoader()}
-     * before being treated as a normal resource.
+     *     Search order is:
      * </p>
+     * <ol>
+     *   <li>{@link ClassLoader#getResource(String) java.lang.Thread.currentThread().getContextClassLoader().getResource(String)}</li>
+     *   <li>{@link ClassLoader#getResource(String) ResourceFactory.class.getClassLoader().getResource(String)}</li>
+     *   <li>{@link ClassLoader#getSystemResource(String) java.lang.ClassLoader.getSystemResource(String)}</li>
+     * </ol>
      *
-     * @param resource Resource as string representation
+     *
+     * @param resource string representation of resource to find in a classloader
+     * @param searchSystemClassLoader true to search {@link ClassLoader#getSystemResource(String)}, false to skip
      * @return The new Resource, or null if string points to a location that does not exist
      * @throws IllegalArgumentException if string is blank
      */
-    default Resource newSystemResource(String resource)
+    default Resource newClassLoaderResource(String resource, boolean searchSystemClassLoader)
     {
         if (StringUtil.isBlank(resource))
             throw new IllegalArgumentException("Resource String is invalid: " + resource);
 
         URL url = null;
-        // Try to format as a URL?
-        ClassLoader loader = Thread.currentThread().getContextClassLoader();
-        if (loader != null)
+
+        List<ClassLoader> classLoaderList = List.of(Thread.currentThread().getContextClassLoader(), ResourceFactory.class.getClassLoader());
+
+        for (ClassLoader classLoader: classLoaderList)
```

Comment:
If you want to include all three sources in the loop, you can do:
```suggestion
        for (Function<String, URL> loader : List.<Function<String, URL>>of(
            Thread.currentThread().getContextClassLoader()::getResource,
            ResourceFactory.class.getClassLoader()::getResource,
            ClassLoader::getSystemResource))
        {
            url = loader.apply(resource);
            if (url == null && resource.startsWith("/"))
                url = loader.apply(resource.substring(1));
            if (url != null)
                break;    
        }
```